motor/discovery/aws/ecs_containers.go

```diff
@@ -0,0 +1,321 @@
+					}
+					for _, task := range taskdescriptions.Tasks {
+						for _, c := range task.Containers {
+							// container assets
+							containersAssets = append(containersAssets, ecsContainerToAsset(ctx, account, region, c, clonedConfig, task))
+						}
+					}
+				}
+			}
+
+			log.Debug().Str("account", account).Str("region", clonedConfig.Region).Int("ecs containers count", len(containersAssets)).Msg("found ecs containers")
+			log.Debug().Str("account", account).Str("region", clonedConfig.Region).Int("ecs container instances count", len(containerInstancesAssets)).Msg("found ecs container instances")
+			res := containersAssets
+			res = append(res, containerInstancesAssets...)
+			return jobpool.JobResult(res), nil
+		}
+		tasks = append(tasks, jobpool.NewJob(f))
+	}
+	return tasks
+}
+
+func mapContainerInstanceState(status *string) asset.State {
+	if status == nil {
+		return asset.State_STATE_UNKNOWN
+	}
+	switch *status {
+	case "REGISTERING":
+		return asset.State_STATE_PENDING
+	case "REGISTRATION_FAILED":
+		return asset.State_STATE_ERROR
+	case "ACTIVE":
+		return asset.State_STATE_ONLINE
+	case "INACTIVE":
+		return asset.State_STATE_OFFLINE
+	case "DEREGISTERING":
+		return asset.State_STATE_STOPPING
+	case "DRAINING":
+		return asset.State_STATE_STOPPING
+	default:
+		return asset.State_STATE_UNKNOWN
+	}
+}
+
+func ecsContainerToAsset(ctx context.Context, account string, region string, c ecstypes.Container, clonedConfig aws.Config, task ecstypes.Task) *asset.Asset {
+	if c.RuntimeId == nil {
+		return nil
+	}
+
+	asset := &asset.Asset{
+		Name:        *c.Name,
+		PlatformIds: []string{containerid.MondooContainerID(*c.RuntimeId)},
+		Platform: &platform.Platform{
+			Kind:    providers.Kind_KIND_CONTAINER,
+			Runtime: providers.RUNTIME_DOCKER_CONTAINER,
+			Family:  []string{*task.PlatformFamily},
+			Version: *task.PlatformVersion,
+		},
+		Connections: []*providers.Config{},
+		Labels:      make(map[string]string),
+		State:       mapContainerState(*c.LastStatus),
+	}
+	containerAttachmentIds := []string{}
+	for _, ca := range c.NetworkInterfaces {
+		containerAttachmentIds = append(containerAttachmentIds, *ca.AttachmentId)
+	}
+	var publicIp string
+	for _, a := range task.Attachments {
+		if stringx.Contains(containerAttachmentIds, *a.Id) {
+			for _, a := range task.Attachments {
+				for _, detail := range a.Details {
+					if *detail.Name == "networkInterfaceId" {
+						publicIp = getPublicIpForContainer(ctx, clonedConfig, *detail.Value)
+					}
+
+					// add connections here
+					asset.Connections = append(asset.Connections, &providers.Config{
+						Backend: providers.ProviderType_SSH, // ???
+						Host:    publicIp,
+					})
+				}
+			}
+		}
+	}
+	asset.Labels[common.IPLabel] = publicIp
+	if publicIp != "" {
+		asset.Name = *c.Name + "-" + publicIp
+	}
+	if c.Image != nil {
+		asset.Labels["image"] = *c.Image
+	}
+	for j := range task.Tags {
```

Comment:
That's nice and I think think the cluster name as well as maybe the task definition would be helpful to identify the origin of the asset easier.